rrs/views.py

```diff
@@ -0,0 +1,1005 @@
+            if recipe_upstream_history:
+                recipe_upstream_all = Raw.get_reup_by_recipes_and_date(
+                        recipe_maintainer_all, recipe_upstream_history.id)
+            else:
+                recipe_upstream_all = Raw.get_reup_by_recipes_and_date(
+                        recipe_maintainer_all)
+
+            for ru in recipe_upstream_all:
+                if ru['status'] == 'Y':
+                    milestone_statistics['up_to_date'] += 1
+                elif ru['status'] == 'N':
+                    if ru['no_update_reason'] == '':
+                        milestone_statistics['not_updated'] += 1
+                    else:
+                        milestone_statistics['cant_be_updated'] += 1
+                else:
+                    milestone_statistics['unknown'] += 1
+
+
+    milestone_statistics['percentage'] = '0'
+    if maintainer_name is None:
+        if milestone_statistics['all'] > 0:
+            milestone_statistics['percentage_up_to_date'] = "%.0f" % \
+                (float(milestone_statistics['up_to_date']) * 100.0 \
+                /float(milestone_statistics['all']))
+            milestone_statistics['percentage_not_updated'] = "%.0f" % \
+                (float(milestone_statistics['not_updated']) * 100.0 \
+                /float(milestone_statistics['all']))
+            milestone_statistics['percentage_cant_be_updated'] = "%.0f" % \
+                (float(milestone_statistics['cant_be_updated']) * 100.0 \
+                /float(milestone_statistics['all']))
+            milestone_statistics['percentage_unknown'] = "%.0f" % \
+                (float(milestone_statistics['unknown']) * 100.0
+                /float(milestone_statistics['all']))
+            if milestone_statistics['all_not_upgraded'] > 0:
+                milestone_statistics['percentage'] = "%.0f" % \
+                    ((float(milestone_statistics['all_upgraded']) * 100.0)
+                    /float(milestone_statistics['all_not_upgraded']))
+        else:
+            milestone_statistics['percentage_up_to_date'] = "0"
+            milestone_statistics['percentage_not_updated'] = "0"
+            milestone_statistics['percentage_cant_be_updated'] = "0"
+            milestone_statistics['percentage_unknown'] = "0"
+    else:
+        if milestone_statistics['all'] > 0:
+            milestone_statistics['percentage'] = "%.0f" % \
+                ((float(milestone_statistics['up_to_date']) /
+                    float(milestone_statistics['all'])) * 100)
+
+    return milestone_statistics
+
+class RecipeList():
+    pk = None
+    name = None
+    version = None
+    summary = None
+    upstream_status = None
+    upstream_version = None
+    outdated = None
+    maintainer_name = None
+    no_update_reason = None
+
+    def __init__(self, pk, name, summary):
+        self.pk = pk
+        self.name = name
+        self.summary = summary
+
+def _get_recipe_list(milestone):
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This function contains 54 lines of code, not including blank lines or lines with only comments, Python punctuation characters, identifiers, or literals. By comparison, 98% of the functions in the CodeGuru reference dataset contain fewer lines of code. Large functions might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this function or break it into multiple functions.